### PR TITLE
feat: add missing typescript-eslint rules

### PR DESCRIPTION
**Pull request by viestat**
This concerns rules added in  [v2.17.0](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#2170-2020-01-20) and [v2.18.0](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#2180-2020-01-27) of https://github.com/typescript-eslint/typescript-eslint/

### CODE REVIEW

File: rules/typescript.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 module.exports = {
   rules: {
     '@typescript-eslint/array-type': ['error', { default: 'array-simple' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
```
Comment (merlinnot) [inline rules/typescript.js]: This is a default, no need to add it explicitly - it's not done for other rules, so I'd keep it consistent: https://github.com/ridedott/eslint-config/blob/58780c910e7022e1b31c7f11adae911d81f4d9aa/index.js#L4

File: rules/typescript.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
     '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],
+    '@typescript-eslint/comma-spacing': 'off',
```
Comment (merlinnot) [inline rules/typescript.js]: Let's add a comment why, you should be able to copy-paste one of the ~ "formatting should be done by Prettier".

File: rules/typescript.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
     '@typescript-eslint/explicit-function-return-type': [
       'error',
       { allowTypedFunctionExpressions: false },
     ],
+    '@typescript-eslint/explicit-module-boundary-types': 'error',
```
Comment (merlinnot) [inline rules/typescript.js]: This is a default, no need to add it explicitly - it's not done for other rules, so I'd keep it consistent: https://github.com/ridedott/eslint-config/blob/58780c910e7022e1b31c7f11adae911d81f4d9aa/index.js#L4

File: rules/typescript.js [hunk 4]
```diff
@@ -15,6 +18,7 @@ module.exports = {
      * TODO (oleg-koval) [2020-01-29] Enable and configure this rule after alignment.
      */
     '@typescript-eslint/naming-convention': 'off',
+    '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',
     '@typescript-eslint/no-dynamic-delete': 'error',
```
Comment (merlinnot) [inline rules/typescript.js]: This is a default, no need to add it explicitly - it's not done for other rules, so I'd keep it consistent: https://github.com/ridedott/eslint-config/blob/58780c910e7022e1b31c7f11adae911d81f4d9aa/index.js#L4

File: rules/typescript.js [hunk 5]
```diff
@@ -47,11 +51,16 @@ module.exports = {
...
+     * This rule has been deprecated and replaced by
+     * explicit-module-boundary-types as of v2.17.0.
+     */
+    '@typescript-eslint/no-untyped-public-signature': 'off',
     '@typescript-eslint/no-unused-vars': [
       'error',
       { ignoreRestSiblings: true },
     ],
+    '@typescript-eslint/prefer-as-const': 'error',
     '@typescript-eslint/prefer-optional-chain': ['error'],
```
Comment (merlinnot) [inline rules/typescript.js]: This is a default, no need to add it explicitly - it's not done for other rules, so I'd keep it consistent: https://github.com/ridedott/eslint-config/blob/58780c910e7022e1b31c7f11adae911d81f4d9aa/index.js#L4

File: rules/typescript.js [hunk 6]
```diff
@@ -47,11 +51,16 @@ module.exports = {
       'error',
       { allowConstantLoopConditions: false, ignoreRhs: true },
     ],
-    '@typescript-eslint/no-untyped-public-signature': 'error',
+    /**
+     * This rule has been deprecated and replaced by
+     * explicit-module-boundary-types as of v2.17.0.
```
Comment (merlinnot) [inline rules/typescript.js]:
```suggestion
     * explicit-module-boundary-types as of @typescript-eslint/* v2.17.0.
```

File: rules/typescript.js (hunk 1 above)
Comment (viestat) [reply rules/typescript.js]: fixed with https://github.com/ridedott/eslint-config/pull/314/commits/8ad156eb1bfc14ba3f60a7cd101ef0207f4422f6

File: rules/typescript.js (hunk 3 above)
Comment (viestat) [reply rules/typescript.js]: fixed with https://github.com/ridedott/eslint-config/pull/314/commits/8ad156eb1bfc14ba3f60a7cd101ef0207f4422f6

File: rules/typescript.js (hunk 4 above)
Comment (viestat) [reply rules/typescript.js]: fixed with https://github.com/ridedott/eslint-config/pull/314/commits/8ad156eb1bfc14ba3f60a7cd101ef0207f4422f6

File: rules/typescript.js (hunk 2 above)
Comment (viestat) [reply rules/typescript.js]: fixed with https://github.com/ridedott/eslint-config/pull/314/commits/8ad156eb1bfc14ba3f60a7cd101ef0207f4422f6

File: rules/typescript.js (hunk 5 above)
Comment (viestat) [reply rules/typescript.js]: fixed with https://github.com/ridedott/eslint-config/pull/314/commits/8ad156eb1bfc14ba3f60a7cd101ef0207f4422f6

**Pull request merged**